client/tx/factory.go

```diff
@@ -130,6 +138,20 @@ func (f Factory) WithFees(fees string) Factory {
 	return f
 }
 
+// WithTips returns a copy of the Factory with an updated tip.
+func (f Factory) WithTips(tip string, tipper string) Factory {
+	parsedTips, err := sdk.ParseCoinsNormalized(tip)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
@atheeshp if you can find a way to:
- move this logic together with the "check tipper no empty" logic
- not panic in the CLI

then that would be ideal! If not, then it's okay like this too.